src/model_scoring_functions.R

```diff
@@ -219,28 +197,48 @@ calc_energy_scores <- function(targets, df_model_output){
                             output_type == "mean") |>
           group_by(clade)
 
-        # Brier score calculation
-        brier_score <- 0
+        # Brier score calculation for the mean
+        brier_point <- 0
         for(k in 1:length(obs_count)){
-          brier_score <- brier_score + obs_count[k]*(df_mean$value[k] - 1)^2 + (N - obs_count[k])*(df_mean$value[k])^2
+          brier_point <- brier_point + obs_count[k]*(df_mean$value[k] - 1)^2 + (N - obs_count[k])*(df_mean$value[k])^2
```

Comment:
vectorize this calculation.

Reply:
Also, consider using a join to merge filtered `df_obs` and `df_model_output` to be 100% certain that the vectors are aligned in terms of all the variables.

Reply:
OR... just merge the oracle output value iwth the model output and scored locations all at once so we don't have to worry about wrangling two data objects over and over again.